website/src/content/service-contract.mdx

```diff
@@ -0,0 +1,78 @@
+
+A feature is marked as Beta when is stable enough to be used in production but is not fully feature complete.
+During the Beta period there might still be issues and planned breaking changes. In this phase commercetools will gather the necessary feedback from consumers of the features in order to finalize the missing things and promote the feature to general availability.
+
+# Hosting
+
+Customers using custom applications are responsible for the maintenance of their custom applications. This includes hosting and deployments.
+commercetools does not take any responsibility towards incidents and service disruptions related to custom applications.
+
+# Releases and breaking changes
+
+Tooling and packages related to custom applications are released and published to [Node Package Manager](https://www.npmjs.com) (NPM) following [Semantic Versioning](https://semver.org/) (SemVer).
+
+Releases are documented in both a `CHANGELOG.md` file and in the GitHub release pages in the following repositories:
+
+- Changelog
+  - [Merchant Center Application Kit](https://github.com/commercetools/merchant-center-application-kit/blob/master/CHANGELOG.md)
+  - [UIKit](https://github.com/commercetools/ui-kit/blob/master/CHANGELOG.md)
+- Release notes
+  - [Merchant Center Application Kit](https://github.com/commercetools/merchant-center-application-kit/releases)
+  - [UIKit](https://github.com/commercetools/ui-kit/releases)
+
+All published packages to NPM are listed in the following package organization:
+
+- [`@commercetools-frontend`](https://www.npmjs.com/org/commercetools-frontend)
+- [`@commercetools-uikit`](https://www.npmjs.com/org/commercetools-uikit)
+
+## Pre-releases and stable releases
+
+By default, all releases go to the `next` distribution tag and should be considered **prereleases**. This gives us a chance to test a release before marking it **stable** in the `latest` distribution channel.
+
+## Breaking changes
+
+Breaking changes are introduced whenever there is a **major** version release, for example from `v1` to `v2`.
+
+> A new major release does not always mean that breaking changes affect consumer of the packages directly.
+> Because of Semantic Versioning, a breaking change can sometimes mean bumping a version of a dependency package
+
+All breaking changes are thoroughly documented in the release notes together with migration steps to facilitate the version update.
+
+# Minimal required maintenance
+
+The Merchant Center applications are always kept up-to-date with the released tools and packages used to build custom applications.
+
+However, customers developing and maintaining custom applications might not always have the time and resources to keep the applications up-to-date.
+commercetools requires that custom applications are at least **maintained at regular periods** and **do not fall back behind too many versions**.
```

Comment:
Can we commit to a major version lifecycle? Saying that each major version will have guaranteed support for 12 or 18 months (meaning no breaking changes in the API), meaning regular maintenance at minimum once a year to upgrade to the latest version.

Reply:
Yeah it's a bit tricky. The breaking changes of the app-kit packages are not the main problem, as they are released as new versions and are documented.
The MC API (breaking) changes is where it gets interesting and where I suppose we need to make some decisions.

Let's make an example: because of feature X, we need to introduce a breaking change in the MC API. The feature is implemented in a (temporary) backwards compatible way and at some point we want to remove the workaround, which is when the breaking change will actually happen. We say that, for example, the deadline will be in 6 months from now.
At the same time, we also release a new version of appkit that somehow includes the new API changes, for example v2.
This means that when the deadline is reached, all apps that have at least v2 will continue to work. So people have like 6 months time to update to at least v2.
However, within those 6 months, appkit can have many more major releases which don't directly relate to the MC API breaking changes.

This describes more or less a possible scenario. We need to put this down into a statement.

Reply:
Yes, to be more specific. When I say breaking changes I mean from the users perspective; when without them changing anything one day their custom application stops working, which as far as I understand should only happen for MC API breaking changes.

If we release a major version v3 on March 1st 2020, then our obligation would be that we won't introduce any MC API breaking changes until September 1st 2020 if 18 months cycle. 18 months would be great since that (with major version updates to AppKit at least every sixth month) would mean that customers can plan to update once a year to the then most recent version of AppKit.

In that time period though, we can make however many new major version increments we want with breaking changes in the library itself (if we should get a more regular release cycle for major releases can be left as a separate discussion and not something that we need to decide on now).

My suggestion would be something like;

"The reason why regular maintenance is important is that the Merchant Center Application Kit relies on internal APIs not guaranteed to be backwards compatible between major versions, which means that an application relying on an old version of appkit at some point might stop working due to breaking changes in the APIs.

Currently, the AppKit major version lifecycle is **18 months from initial release** with guarantees for no breaking changes in internal MC APIs. This mean that custom applications need to be updated to the most recent version of AppKit **at minimum once per year**."

Reply:
I'm not an expert but 18 months seems quite a lot to me. Is this like a standard?

> regular maintenance to be updated to the most recent version of AppKit at minimum once per year

I would suggest to have it more frequent, like once a quarter or twice a year. At least for now, especially during the beta phase. WDYT?

Reply:
I hearded we settled on a year, right? A year subjectively feels quite long but is reasonable I think. From and MC engineer perspective a year looks like about 12 breaking releases (given the past year). From a customer perspective that is not tolerable. Striking a reasonable and working balance helping both sides important. Saying this, as what we promise as a maintainance window can potentially in turn slow us down internally (which is fine but worth mentioning).

As stated above, the only breaking change affecting prior released versions is a pairing between our MC API and App-Kit and its interplay. This is where we should be as backwards compatible as possible at all times. Note also that I think in the past we ironed out quite a few obvious breaking changes so we might naturally ending up at lower cadence in the future. 

I think we can maybe give recommendations while stating the maintainance window.

> To be able to utilise the latest improvements, fixes and security updates (also by used libraries) to our Custom Applications SDK we recommend updating it regularily. This can be accomplished through automation (e.g. Renocate, Greenkeeper) or manual schedules. In general, updating the SDK once a quarter is recommended.

We should also note that we have the two mentioned distribution channels (which is quite normal in in the npm world). There we can say that releases the `next` are seen as experimental and not meant for external production use and mostly for us to ensure stability. Then only what's released under `latest` is falling into our maintenance window. Note, that you have to explicitely opt-into consuming releases from `next`. It's nothing which "just happens".

What we could also note that is that we do not backport security fixes or general bugs to older versions. So if somebody is on v9.23.4 while the last version is v12.10.1 with a important fix with v12.10.2 we will not backport or create a fix via v9.23.4. Which why we recommend staying not too far behind.

We can also plot a graph with the releases lines etc for that. Would it be okay if I just open a PR to this one proposing some changes. Just as a basis cause otherwise this might go on longer than it has to :)

Reply:
Please feel free to improve the docs in this PR.

Reply:
I will also pick this up again later this week